src/components/Video.cpp

```diff
@@ -167,6 +167,12 @@ bool Video::setGeometry(unsigned width, unsigned height, unsigned maxWidth, unsi
       return false;
     }
   }
+  else if (_vertexArray)
```

Comment:
While this appears correct, I think we should avoid handling `_vertexArray` here directly. The `ensure*` methods were intended to completely manage the opengl objects to avoid ad-hoc state changes throughout the code.

In the case of `_vertexArray`, the correct entrypoint to change it would be `ensureView`, which is always executed when the core calls the refresh callback.

- In `ensureView`, the `if` that checks whether the view needs to be rebuilt could be updated to force a rebuild when we've changed from hardware to non-hardware rendering (either by adding more state to `Video` or using `!_hw.enabled && _vertexArray != 0`).
- Then, in `ensureVertexArray`, we can delete the vertex array if `!_hw.enabled && _vertexArray = 0`.

I think this approach would be cleaner: it simply rebuilds the view when we change from hardware to non-hardware rendering.

Reply:
I've moved the changes back into `ensureVertexArray`.

Reply:
`ensureView` bails early if nothing changed regarding the view, so I think it's possible, though unlikely, for the current solution to not delete the vertex array. This would happen if the new non-hw core started running with the exact same resolution and rotation as the previous hw core.

I think this can be solved by having `ensureFramebuffer`, which is called in `setGeometry`, not bail early if `!_hw.enabled && _hw.frameBuffer != 0` (changed from hw to non-hw). The opposite check is already there. When `ensureFramebuffer` changes stuff, it also forces the view to be rebuilt, which ensures the vertex array will be deleted.